src/AppKit/NSView.cs

```diff
@@ -31,20 +33,25 @@ sealed class SortData {
 #endif
 		static nint view_compare (IntPtr view1, IntPtr view2, IntPtr context)
 		{
-			var data = (SortData) GCHandle.FromIntPtr (context).Target;
+			var data = GCHandle.FromIntPtr (context).Target as SortData;
 			try {
-				var a = (NSView) Runtime.GetNSObject (view1);
-				var b = (NSView) Runtime.GetNSObject (view2);
-				return (nint) (long) data.Comparer (a, b);
+				if (Runtime.GetNSObject (view1) is NSView a
+						&& Runtime.GetNSObject (view2) is NSView b) {
+					var result = data?.Comparer?.Invoke (a, b);
+					return result is null ? (nint) (long) NSComparisonResult.Same : (nint) (long) result;
+				} else {
+					return (nint) (long) NSComparisonResult.Same;
+				}
 			} catch (Exception e) {
-				data.Exception = e;
+				if (data is not null)
+					data.Exception = e;
 				return (nint) (long) NSComparisonResult.Same;
 			}
 		}
 
-		public unsafe void SortSubviews (Func<NSView, NSView, NSComparisonResult> comparer)
+		public unsafe void SortSubviews (Func<NSView, NSView, NSComparisonResult>? comparer)
```

Comment:
`comparer` shouldn't be null according to the code.

However, the two arguments to compare should allow null:

```suggestion
		public unsafe void SortSubviews (Func<NSView?, NSView?, NSComparisonResult> comparer)
```